### ISSUE TITLE
file repository: arrows for ease of navigation between images?

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#2405]
I just assigned contexts in Kiosk to a bunch of images and I found myself really wanting arrows to be able to navigate between them more easily. Was this discussed and decided against for some reason? If one has to look through a bunch of images, it seems easier to just be able to navigate through them that way rather than having to exit out and click on the next one all the time.

**Comment by urapadmin** [comment]
no, that has not been discussed so far. That would be an image box and I had indeed contemplated it for View for one of the next iterations.

**Comment by luizaogs** [comment]
For me, at least, it would be very useful.

**Comment by luizaogs** [comment]
ho ho ho

**Comment by luizaogs** [comment]
Writing a report for MKAP currently and again wishing for arrows. Just letting you know that this wasn't just a fleeting desire prompted by insanity while in Egypt.

**Comment by urapadmin** [comment]
oh, I never thought so. The request is valid. I have too many valid requests 😬

**Comment by urapadmin** [comment]
I would love to hear of other obstacles during that process. Are you using Q&V?
do you want full text search?

**Comment by luizaogs** [comment]
This is just a preliminary publication and I had already written a report about my units while in the field, so currently I'm mostly translating it for public consumption rather than looking through the data in a lot of detail again. So I haven't had a need for Q&V yet. 

But I am trying to decide on what images to publish, hence this request coming back. I am tagging images as JARCE2024 in the repository to indicate that I want to use specific ones in the article, which is very useful.

**Comment by urapadmin** [comment]
okay. Then FTS isn't anything you'd need, either. Let me know if I can do anything to help (should you want a query, that would be so wonderful :o). 
But at the very least I won't switch off Kiosk right now (PostgreSQL update would take a few hours).

**Comment by luizaogs** [comment]
Ha after a season I don't think we have enough to really query yet :)

But yes, please don't switch it off!

**Comment by urapadmin** [comment]
As for IIIFe support, which I would want to build into this, see http://openseadragon.github.io/

**Comment by urapadmin** [comment]
also getting @lbestock on board here.

Arrows in the file edit dialog are a fickle thing to do for me. I'd rather not turn that old code base into anything new.  
I have an idea, but I am not sure it would serve your needs:

I could allow starring files in the file repository (you know, like for bulk tagging) and then switch to a full image view that takes place right in the file repository (so we are not talking about the popup here). It might even be triggered with the thumbnail size selector.
 
The full image view would show the image in its actual extend. I can also imagine to offer a zoom there (as for #2412). 
In that mode you would get back and forth buttons to step through the starred images. 
The caveat: You would see about the same data for the image that you know from the multi-image view in the file repository AND you would not be able to change data. That would need the usual click on the image to get into the file edit dialog.
But at least you would be able to a) step through a bunch of images and b) see them large.

**Comment by luizaogs** [comment]
I don't know if it exactly serves the needs I had in the field, because I would first have to know what I'm selecting to get to this - and moving through large amounts of images was the question/issue. But it sounds helpful for looking through a selection of images, and if you don't want to mess with the old code more generally I think it can only help.

**Comment by luizaogs** [comment]
Though I guess actually, if I wanted to see everyone on a specific page I could just use the full star button? So yes, I think it would actually serve the needs I had (since I could just use the full star button for every page). Or am I misunderstanding?

**Comment by lbestock** [comment]
I think this will be useful. I'm slightly on the fence about the needing to star things to begin with rather than just being able to scroll through and whether or not that step is an impediment because I will forget the option exists. But it's definitely better than always going back to the search as one now does. I think this is different than changing data, and so this part doesn't strike me as a problem at all. (I use the FR all the time. I almost never am changing data in there, just searching and usually using my eyes to do a final sort, not the data.) But I also know I don't use the "medium thumbnails" nearly as often as I should and I do wonder if that would solve half my problem on this front, and if having a "large thumbnails" would solve all of it. 

But the ability to zoom changes it again - I might be able to find what I'm looking for with larger thumbnails, but actually looking at it without a download, using IIIF, yes please.

**Comment by urapadmin** [comment]
@luizaogs I should have learned in my life never to give out Christmas vouchers. I Never got even close to this. This is rather post-season, right? It would be a bit rushed before MKAP. But I might still be able to make post-season work a bit easier with it.

**Comment by luizaogs** [comment]
I think my real need for it was with MADP, actually. So I would say whenever you can get to it would be great - it would definitely make sorting through images post-season easier.

**Comment by lbestock** [comment]
Hanukkah and Christmas are at the same time this year, if you get back to thinking about holiday vouchers :)
A thought though, because when I last wrote about the topic we must not have been doing BUAP stuff yet. And for that, the inability to change data would be a real downside. There one wants to see the image full enough to read the signboard, then assign the context while looking at the board. Clicking through a bunch and doing this all at once would be very helpful in that case.

(Strikes me that a few swirling things right now are about adding data that is not properly field recording. This, comparanda, interpretations....)

**Comment by luizaogs** [comment]
You said above that this is something you had thought of for Q&V, though I had asked for it specifically in the file repository. Right now I'm using Q&V and also wanting arrows. Just letting you know...

**Comment by urapadmin** [comment]
I agree that this needs to be some sort of versatile image viewer that can pop up everywhere where images and lists of images are involved. It should also be able to deal with IIIFs and perhaps even needs to intersect with the image annotation tool we need for #1982 and co.
So a project in and of itself and hence not happening too quickly. So hang in there.

**Comment by urapadmin** [comment]
I might want to consider this or a "preview" of some sort earlier than the real thing to support the Buap / Abydos workflow

**Comment by lbestock** [comment]
Implement #929 first. It might or might not make it much less needed.

**Comment by urapadmin** [comment]
now scheduled to be in Kiosk 1.7

**Comment by urapadmin** [comment]
### Kiosk 1.6.17
- the new file viewer has arrow buttons in the upper right corner
- with those you step through all images on the current page of the file repository (but not to the next or previous page should there be pagination active)

**Comment by luizaogs** [comment]
The arrows are incredibly helpful.
